android/lib/src/main/java/com/ern/api/impl/navigation/ElectrodeNavigationFragmentDelegate.java

```diff
@@ -283,12 +283,20 @@ protected LaunchConfig createNextLaunchConfig(@NonNull Route route) {
         return config;
     }
 
+    /**
+     * This method should return the fragment that is hosting the current RN view.
+     *
+     * @return Fragment
+     */
     @Nullable
     protected Fragment getTopOfTheStackFragment() {
         if (mFragment.getActivity() != null) {
             List<Fragment> fragments = mFragment.getActivity().getSupportFragmentManager().getFragments();
-            if (fragments.size() > 0) {
-                return fragments.get(fragments.size() - 1);
+            for (int i = fragments.size() - 1; i >= 0; i--) {
```

Comment:
If you prefer to use the inbuilt utility function `FragmentManager` utility methods
- `getBackStackEntryAt` & `getBackStackEntryCount - 1`, `getBackStackEntryAt` can give the name of the Fragment that we used when `FragmentTransaction.addToBackStack(String)`
-  & then `findFragmentByTag` to get the fragment

Approving it, whichever you prefer :)

Reply:
The TAG is something that may change from an activity level. This delegate will not know what tag to look for. Also, when it comes to a dialog(overlays) it needs to check for    `f instanceof NavigationRouteHandler`

Reply:
well at some point you'd be [addToBackStack](https://github.com/electrode-io/ernnavigation-api-impl-native/blob/0928967b8a7d6c765af6ce777873bfe4099b441c/android/lib/src/main/java/com/ern/api/impl/core/ElectrodeBaseActivityDelegate.java#L146) with the fragment component name. `getBackStackEntryAt` will give you the same fragment back.
findFragmentByTag will give you the fragment that you can compare `instanceof`
As I said, it's using utility functions if you'd prefer.